src/app/portlets/dot-edit-page/content/components/dot-edit-page-workflows-actions/dot-edit-page-workflows-actions.component.ts

```diff
@@ -0,0 +1,40 @@
+import { Component, OnInit, Input, OnChanges } from '@angular/core';
+import { MenuItem } from 'primeng/primeng';
+import { Workflow } from '../../../../../shared/models/workflow/workflow.model';
+import { WorkflowService } from '../../../../../api/services/workflow/workflow.service';
+
+@Component({
+    selector: 'dot-edit-page-workflows-actions',
+    templateUrl: './dot-edit-page-workflows-actions.component.html',
+})
+export class DotEditPageWorkflowsActionsComponent implements OnInit, OnChanges {
+
+    @Input() Inode: string;
+    @Input() label: string;
+
+    pageWorkflows: Workflow[];
+    workflowsActions: MenuItem[] = [];
+
+    constructor(private workflowsService: WorkflowService) {}
+
+    ngOnInit() {
+        this.workflowsService.getPageWorkflows(this.Inode)
+            .subscribe((data) => {
+                this.pageWorkflows = data;
```

Comment:
Why do we need this one? can you just do:

```this.workflowsActions = this.getWorkflowOptions(data);```

?

Reply:
after we get the data from the service, it needs to be transform so the split button component gets the right data, so implementing the way you suggest won't let you transform the data